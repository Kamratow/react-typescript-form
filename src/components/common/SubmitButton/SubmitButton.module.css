.SubmitButtonWrapper {
    margin-top: 30px;
    width: 100%;
}

.SubmitButton {
    display: block;
    border-radius: 15px;
    margin: auto;
    background-color: #000;
    color: #fff;
    cursor: pointer;
    font-size: 25px;
    padding: 15px 25px;
    border: 3px solid #000;
}

.SubmitButton:focus {
    outline: none;
    box-shadow: 0 0 6pt #008000;
    border-color: #008000;
}

.SubmitButton:hover:enabled {
    background-color: #fff;
    color: #000;
}

.SubmitButton:disabled {
    background-color: #a9a9a9;
    cursor: not-allowed;
    border-color: #a9a9a9;
}
