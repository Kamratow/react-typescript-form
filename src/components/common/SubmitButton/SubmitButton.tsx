import React from "react";

import styles from "./SubmitButton.module.css";

export interface ISubmitButtonProps {
    buttonLabel: string;
    disabled: boolean;
}

const SubmitButton: React.FunctionComponent<ISubmitButtonProps> = ({
    buttonLabel,
    disabled,
}) => {
    return (
        <div className={styles.SubmitButtonWrapper}>
            <button
                className={styles.SubmitButton}
                type="submit"
                disabled={disabled}
            >
                {buttonLabel}
            </button>
        </div>
    );
};

export default SubmitButton;
