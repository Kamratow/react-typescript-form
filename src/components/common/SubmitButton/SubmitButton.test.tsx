import React from "react";
import { shallow } from "enzyme";

import SubmitButton, { ISubmitButtonProps } from "./SubmitButton";

const defaultProps: ISubmitButtonProps = {
    disabled: false,
    buttonLabel: "test button label",
};

describe("SubmitButton", () => {
    it("should render properly with default props", () => {
        const wrapper = shallow(<SubmitButton {...defaultProps} />);

        expect(wrapper).toMatchSnapshot();
    });
});
