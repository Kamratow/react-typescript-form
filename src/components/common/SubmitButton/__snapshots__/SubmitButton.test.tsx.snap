// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton should render properly with default props 1`] = `
<div
  className="SubmitButtonWrapper"
>
  <button
    className="SubmitButton"
    disabled={false}
    type="submit"
  >
    test button label
  </button>
</div>
`;
